homeassistant/components/upcloud/__init__.py

```diff
@@ -171,7 +299,7 @@ def device_class(self):
     def device_state_attributes(self):
         """Return the state attributes of the UpCloud server."""
         return {
-            x: getattr(self.data, x, None)
+            x: getattr(self._server, x, None)
             for x in (
                 ATTR_UUID,
```

Comment:
Why put uuid in attr?

Reply:
or title, hostname ? 

What is state ?

Reply:
Especially since state is already returned as `state`

Reply:
This isn't something new to this PR, but sure, we can tweak it while at it if you like.

Anyway I suppose state could be removed if we want, although there are some "raw" states that we don't have a mapping for in the dict, and thus they're returned as None. Not sure if we should have all of these there instead/additionally.

uuid, title, and hostname to me seem as useful as any info we have there to have in device state attributes. uuid/title can be in the entity name, depends, so those are kind of redundant.

Reply:
Passing unmapped states through as is now, removed raw state from state attributes.